core/src/main/java/org/apache/cxf/io/CachedOutputStream.java

```diff
@@ -225,7 +225,11 @@ public void close() throws IOException {
             ciphers.clean();
         }
         if (!maybeDeleteTempFile(currentStream)) {
-            postClose();
+            try {
```

Comment:
@Sevyls thank you for the investigation and the consequent suggested fix. I think we should attack the problem from the different angle (as you also mentioned in the ticket):
 - `postClose()` could raise `IOException` so does `close()` and this is legitimate, no need to catch it
 - however, as you found out, `CacheAndWriteOutputStream` may attempt to close the underlying stream many times
I think the proper solution would be:
 - `CacheAndWriteOutputStream`  should keep track if the `flowThroughStream` has been closed already 
 - `closeFlowthroughStream()` and `postClose()` should set or/and check if the `flowThroughStream` has been closed already and behave appropriately (flush/close or do nothing)

What do you think?